File: Data Structures & Algorithms/permutations/submission-0.py
class Solution:
    def permute(self, nums: List[int]) -> List[List[int]]:
        results = []
        self.recursive(nums, [], results)        
        return results

    def recursive(self,
        nums: List[int],
        perm: List[int],
        results: List[List[int]]):

        # Base case
        if len(nums) == 0:
            results.append(perm.copy())
            return

        for i, num in enumerate(nums):
            perm.append(num)

            other_nums = nums[:i] + nums[i+1:]
            self.recursive(other_nums, perm, results)

            perm.pop()

File: Data Structures & Algorithms/missing-number/submission-0.go
func missingNumber(nums []int) int {
    n := len(nums)
    
    xor := 0
    for i := 0; i <= n; i++ {
        xor ^= i
    }

    for _, num := range nums {
        xor ^= num
    }

    return xor
}
